test/dhkem_test.inc

```diff
@@ -0,0 +1,707 @@
+    0x58, 0x4a, 0x72, 0x6b, 0x9c, 0x17, 0x67, 0x96,
+    0xca, 0xd0, 0x16, 0x9b, 0xa4, 0x09, 0x33, 0x32,
+    0xcb, 0xd2, 0xdc, 0x3a, 0x9f
+};
+
+static const unsigned char p521_auth_ikms_priv[] = {
+    0x00, 0x10, 0x18, 0x58, 0x45, 0x99, 0x62, 0x5f,
+    0xf9, 0x95, 0x3b, 0x93, 0x05, 0x84, 0x98, 0x50,
+    0xd5, 0xe3, 0x4b, 0xd7, 0x89, 0xd4, 0xb8, 0x11,
+    0x01, 0x13, 0x96, 0x62, 0xfb, 0xea, 0x8b, 0x65,
+    0x08, 0xdd, 0xb9, 0xd0, 0x19, 0xb0, 0xd6, 0x92,
+    0xe7, 0x37, 0xf6, 0x6b, 0xea, 0xe3, 0xf1, 0xf7,
+    0x83, 0xe7, 0x44, 0x20, 0x2a, 0xaf, 0x6f, 0xea,
+    0x01, 0x50, 0x6c, 0x27, 0x28, 0x7e, 0x35, 0x9f,
+    0xe7, 0x76
+};
+
+static const unsigned char p521_auth_expected_enc[] = {
+    0x04, 0x01, 0x7d, 0xe1, 0x2e, 0xde, 0x7f, 0x72,
+    0xcb, 0x10, 0x1d, 0xab, 0x36, 0xa1, 0x11, 0x26,
+    0x5c, 0x97, 0xb3, 0x65, 0x48, 0x16, 0xdc, 0xd6,
+    0x18, 0x3f, 0x80, 0x9d, 0x4b, 0x3d, 0x11, 0x1f,
+    0xe7, 0x59, 0x49, 0x7f, 0x8a, 0xef, 0xdc, 0x5d,
+    0xbb, 0x40, 0xd3, 0xe6, 0xd2, 0x1d, 0xb1, 0x5b,
+    0xdc, 0x60, 0xf1, 0x5f, 0x2a, 0x42, 0x07, 0x61,
+    0xbc, 0xae, 0xef, 0x73, 0xb8, 0x91, 0xc2, 0xb1,
+    0x17, 0xe9, 0xcf, 0x01, 0xe2, 0x93, 0x20, 0xb7,
+    0x99, 0xbb, 0xc8, 0x6a, 0xfd, 0xc5, 0xea, 0x97,
+    0xd9, 0x41, 0xea, 0x1c, 0x5b, 0xd5, 0xeb, 0xee,
+    0xac, 0x7a, 0x78, 0x4b, 0x3b, 0xab, 0x52, 0x47,
+    0x46, 0xf3, 0xe6, 0x40, 0xec, 0x26, 0xee, 0x1b,
+    0xd9, 0x12, 0x55, 0xf9, 0x33, 0x0d, 0x97, 0x4f,
+    0x84, 0x50, 0x84, 0x63, 0x7e, 0xe0, 0xe6, 0xfe,
+    0x9f, 0x50, 0x5c, 0x5b, 0x87, 0xc8, 0x6a, 0x4e,
+    0x1a, 0x6c, 0x30, 0x96, 0xdd
+};
+
+static const unsigned char p521_auth_expected_secret[] = {
+    0x26, 0x64, 0x8f, 0xa2, 0xa2, 0xde, 0xb0, 0xbf,
+    0xc5, 0x63, 0x49, 0xa5, 0x90, 0xfd, 0x4c, 0xb7,
+    0x10, 0x8a, 0x51, 0x79, 0x7b, 0x63, 0x46, 0x94,
+    0xfc, 0x02, 0x06, 0x1e, 0x8d, 0x91, 0xb3, 0x57,
+    0x6a, 0xc7, 0x36, 0xa6, 0x8b, 0xf8, 0x48, 0xfe,
+    0x2a, 0x58, 0xdf, 0xb1, 0x95, 0x6d, 0x26, 0x6e,
+    0x68, 0x20, 0x9a, 0x4d, 0x63, 0x1e, 0x51, 0x3b,
+    0xad, 0xf8, 0xf4, 0xdc, 0xfc, 0x00, 0xf3, 0x0a
+};
+
+static const TEST_DERIVEKEY_DATA ec_derivekey_data[] = {
+    {
+      "P-256",
+      p256_ikme, sizeof(p256_ikme),
+      p256_ikme_pub, sizeof(p256_ikme_pub),
+      p256_ikme_priv, sizeof(p256_ikme_priv)
+    },
+    {
+      "P-256",
+      p256_ikmr, sizeof(p256_ikmr),
+      p256_ikmr_pub, sizeof(p256_ikmr_pub),
+      p256_ikmr_priv, sizeof(p256_ikmr_priv)
+    },
+    {
+      "P-521",
+      p521_ikme, sizeof(p521_ikme),
+      p521_ikme_pub, sizeof(p521_ikme_pub),
+      p521_ikme_priv, sizeof(p521_ikme_priv)
+    }
+};
+
+static const TEST_ENCAPDATA ec_encapdata[] = {
+    {
+        "P-256",
+        p256_ikme, sizeof(p256_ikme),
+        p256_ikmr_pub, sizeof(p256_ikmr_pub),
+        p256_ikmr_priv, sizeof(p256_ikmr_priv),
+        p256_expected_enc, sizeof(p256_expected_enc),
+        p256_expected_secret, sizeof(p256_expected_secret),
+    },
+    {
```

Comment:
I would suggest adding the auth test vector for P256 from the RFC also.

Reply:
I left it out since the HPKE tests are likely to rehit all of theses cases... I just did p521 as it is the largest data set, so the combos of p-256 non auth, p-521 non auth and p521 auth should be hitting all the bits of code that actually matter..